src/App.tsx

```diff
@@ -1,20 +1,55 @@
 import { SearchHeader } from "./components/header/SearchHeader";
-import { ClothingCardsGrid } from "./components/clothingGrid/ClothingCardsGrid";
+import { ProductsGrid } from "./components/productsGrid/ProductsGrid";
 import { Sidebar } from "./components/sidebar/Sidebar";
 import { useState } from "react";
+import { useDebounce } from "./hooks/useDebounce";
 
 function App() {
   const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+  const [sortValues, setSortValues] = useState({
+    sortBy: "",
+    sortOrder: "",
+  });
+  const { sortBy, sortOrder } = sortValues;
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+  const [isSortModalOpen, setIsSortModalOpen] = useState(false);
+
+  const handleOpenSortModal = () => {
+    setIsSortModalOpen((prev) => !prev);
+  };
 
   const handleOpenSidebar = () => {
     setIsSidebarOpen((prev) => !prev);
   };
 
+  const handleChangeSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    setSearchTerm(value);
+  };
+
+  const handleChangeSort = (event: React.MouseEvent<HTMLButtonElement>) => {
+    const { value, name } = event.currentTarget;
+    setSortValues({ sortBy: name, sortOrder: value });
+    setIsSortModalOpen(false);
+  };
+
   return (
-    <main className="flex flex-col gap-8 bg-white ">
+    <main className="flex flex-col min-h-screen w-full gap-8 bg-white ">
       <Sidebar handleOpenSidebar={handleOpenSidebar} isOpen={isSidebarOpen} />
-      <SearchHeader handleOpenSidebar={handleOpenSidebar} />
-      <ClothingCardsGrid />
+      <SearchHeader
```

Comment:
suggestion: rename to SearchAndSortHeader or separate into two different components. So that it is explicit the responsibility of the unique component or each component